### ISSUE TITLE
SSL certificate problem: self signed certificate

### ISSUE THREAD

**Issue by jlaustill** [algolia/algoliasearch-wordpress#512]
### What did you expect to happen? 
I expected indexing to work

### What happened instead?
I get this error 
WP_Error Object
(
    [errors] => Array
        (
            [http_request_failed] => Array
                (
                    [0] => SSL certificate problem: self signed certificate
                )

        )

    [error_data] => Array
        (
        )

)
### How can we reproduce this behavior?
I am trying to run this on my local dev box.  I'm not getting this error in production or qa where we have legit ssl certs.  I have have searched high and low and I can't find a way to make this work on my local dev instance.  Is there an option I'm just overlooking?  This is super annoying.

Can you provide a link to a page which shows this issue?

### Technical info
* WordPress version: 4.5.4
* Algolia Search plugin version: 1.7.0

**Comment by jlaustill** [comment]
adding these lines to wp-config.php seems to get it to work.  Is this a bad idea?

add_filter('https_ssl_verify', '__return_false');

add_filter('https_local_ssl_verify', '__return_false');

**Comment by rayrutjes** [comment]
Hi @jlaustill , 

The error you get is indeed related to the fact that cURL is not able to validate your certificate.

You can indeed deactivate that check in development with the snippet you provided.

I wouldn't recommend disabling certificate checks on production environments though, where you should focus on obtaining a valid certificate.

**Comment by jlaustill** [comment]
Absolutely never do this in production :)  I ran into another possible fix that involved adding your self signed cert to your local store.  I'm going to test that and if it works I'll probably write up a blog post on these two solutions.  I'd love to contribute back if there's a good place for this information in the docs?  I don't think this is really a bug, but I did waste an entire day on it so far just figuring it out :(
